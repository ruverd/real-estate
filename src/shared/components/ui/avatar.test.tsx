

import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Avatar, AvatarFallback, AvatarImage } from "./avatar";

const mockAvatarUrl = "https://github.com/ruverd.png";

const renderComponent = ({ avatar }: { avatar?: string } = {}) => {
  return render(
    <Avatar>
      {avatar && <AvatarImage src={avatar} alt="test user" />}
      <AvatarFallback>TU</AvatarFallback>
    </Avatar>
  );
};

describe("Avatar", () => {
  it("should match snapshot without image", () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot with image", () => {
    const { container } = renderComponent({ avatar: mockAvatarUrl });
    
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render avatar without image", () => {
    renderComponent();

    expect(screen.getByText("TU")).toBeInTheDocument();
  });
});
