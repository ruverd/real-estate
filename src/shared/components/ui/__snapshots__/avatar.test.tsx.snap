// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should match snapshot with image 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
>
  <span
    class="bg-muted flex size-full items-center justify-center rounded-full"
    data-slot="avatar-fallback"
  >
    TU
  </span>
</span>
`;

exports[`Avatar > should match snapshot without image 1`] = `
<span
  class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
  data-slot="avatar"
>
  <span
    class="bg-muted flex size-full items-center justify-center rounded-full"
    data-slot="avatar-fallback"
  >
    TU
  </span>
</span>
`;
